common/src/main/java/tachyon/security/authentication/AuthenticationProvider.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed to the University of California, Berkeley under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.security.authentication;
+
+import javax.security.sasl.AuthenticationException;
+
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+
+import tachyon.Constants;
+import tachyon.conf.TachyonConf;
+
+public interface AuthenticationProvider {
+  /**
+   * Factory method to create an instance of {@link AuthenticationProvider} based on
+   * {@link AuthType} and {@link TachyonConf}.
+   */
+  class Factory {
+    public static AuthenticationProvider getAuthenticationProvider(AuthType authType,
+        TachyonConf conf) throws AuthenticationException {
+      switch (authType) {
+        case SIMPLE:
+          return new SimpleAuthenticationProviderImpl();
+        case CUSTOM:
+          Preconditions.checkState(conf
+              .containsKey(Constants.SECURITY_AUTHENTICATION_CUSTOM_PROVIDER));
```

Comment:
Don't think this is necessary, since the next call will throw the exception if it doesn't exist.